js/form.js

```diff
@@ -0,0 +1,66 @@
+import {picturesContainer} from './renderingPicture.js';
+
+const VALID_SYMBOLS = /^#[a-zа-яё0-9]{1,19}$/i;
+const imgUploadForm = document.querySelector('.img-upload__form');
+const fileFild = picturesContainer.querySelector('.img-upload__input');
+const filtersForm = picturesContainer.querySelector('.img-upload__overlay');
+const battonHideFiltorsForm = filtersForm.querySelector('.img-upload__cancel');
+const inputTag = filtersForm.querySelector('.text__hashtags');
+const inputComment = filtersForm.querySelector('.text__description');
+
+const pristine = new Pristine(imgUploadForm,{
+  classTo: '.img-upload__field-wrapper',
+  errorTextParent: '.img-upload__field-wrapper',
+  errorTextClass: '.img-upload__field-wrapper',
+});
+const focusForms = (evt)=>{
+  if(evt.key === 'Escape'){
+    evt.stopPropagation();
+  }
+};
+const isValidateTag = (tag)=> VALID_SYMBOLS.test(tag);
+const isUniqueValue = (tags) =>{
+  const tagsTolowerCase = tags.map((tag) => tag.tagsTolowerCase());
+  return tagsTolowerCase.length === new Set(tagsTolowerCase).size;
+};
+const isLengthTag = (tag) => tag.length <= 5;
+
+const validateTags = (value)=>{
+  const tags = value
+    .trim()
+    .split(' ')
+    .filter((tag) => tag.trim().length);
+  return isUniqueValue(tags) && tags.every(isValidateTag) && isLengthTag(tags);
+};
+
+pristine.addValidator(
+  inputTag,
+  validateTags,
+  'ХешТаг не соответствует требованиям',
+);
+const hideModal = (e) => {
+  if (e.key && e.key !== 'Escape') {
+    return;
+  }
+  filtersForm.classList.add('hidden');
+  e.preventDefault();
+  document.body.classList.remove('modal-open');
+  battonHideFiltorsForm.removeEventListener('click',hideModal);
+  battonHideFiltorsForm.removeEventListener('keydown',hideModal);
+  inputTag.removeEventListener('keydown',focusForms);
+  inputComment.removeEventListener('keydown',focusForms);
+  imgUploadForm.reset();
+};
+
+const showModal = ()=>{
+  filtersForm.classList.remove('hidden');
+  document.body.classList.add('modal-open');
+  battonHideFiltorsForm.addEventListener('click',hideModal);
+  document.addEventListener('keydown',hideModal);
+  inputTag.addEventListener('keydown',focusForms);
+  inputComment.addEventListener('keydown',focusForms);
+};
+
+fileFild.addEventListener('change',showModal);
+
+
```

Comment:
Нужно и на отправку вешать ивент, чтобы блокировать отправку, если хоть валидация не пройдена
```suggestion
imgUploadForm.addEventListener('submit', (e) => {
  if (!pristine.validate()) {
    e.preventDefault();
  }
});

```